birdset/datamodule/beans_datamodule.py

```diff
@@ -104,4 +106,29 @@ def label_to_id_fn(batch):
             ),
         )
         
-        return dataset
\ No newline at end of file
+        return dataset
+    
+    
+    def _preprocess_data(self, dataset):
+        """
+        Preprocess the data. This calls the _ksamples function to select k samples per class and calls the embedding extraction function. If multilabel is the task we will one hot encode. 
+        """
+
+        # Check if actually a dict
+        dataset = self._ksamples(dataset)
+        dataset = self._compute_embeddings(dataset)
```

Comment:
@XgamerTV I would suggest to first computing the embeddings and then create the k_samples. We then can cache the embeddings and avoid recomputing them for every experiment.

I would suggest tackling this in another issue.

Reply:
See #9